Src/GoogleApis.Auth.DotNet4.Tests/OAuth2/DefaultCredentialProvider/DefaultCredentialProviderTests.cs

```diff
@@ -0,0 +1,217 @@
+            credentialProvider.SetFileContents(credentialFilepath, DummyUserCredentialFileContents);
+
+            ICredential credential = credentialProvider.GetApplicationDefaultCredential();
+
+            if (!(credential is UserCredential))
+            {
+                Assert.Fail(string.Format("Expected credential of type UserCredential. Obtained {0} instead.", credential.GetType()));
+            }
+        }
+
+        [Test]
+        public void TestGetApplicationDefaultCredentials_UserCredentials_FromWellKnownFileLocation()
+        {
+            TestDefaultCredentialProvider credentialProvider = new TestDefaultCredentialProvider();
+            credentialProvider.SetFileContents(WellKnownCredentialFilePath, DummyUserCredentialFileContents);
+
+            ICredential credential = credentialProvider.GetApplicationDefaultCredential();
+
+            if (!(credential is UserCredential))
+            {
+                Assert.Fail(string.Format("Expected credential of type UserCredential. Obtained {0} instead.", credential.GetType()));
+            }
+        }
+
+        #endregion
+
+        #region ServiceAccountCredential
+
+        [Test]
+        public void TestGetApplicationDefaultCredentials_ServiceAccountCredentials_FromEnvironmentVariable()
+        {
+            string credentialFilepath = "TempFilePath.json";
+            TestDefaultCredentialProvider credentialProvider = new TestDefaultCredentialProvider();
+            credentialProvider.SetEnvironmentVariable(CredentialEnvironmentVariable, credentialFilepath);
+            credentialProvider.SetFileContents(credentialFilepath, DummyServiceAccountCredentialFileContents);
+
+            ICredential credential = credentialProvider.GetApplicationDefaultCredential();
+
+            if (credential is IScopableCredential)
+            {
+                IScopableCredential scopableCredential = credential as IScopableCredential;
+                if (scopableCredential.IsCreateScopedRequired)
+                {
+                    credential = scopableCredential.CreateScoped(new[] { "https://www.googleapis.com/auth/cloud-platform" });
+                }
+            }            
+
+            if (!(credential is ServiceAccountCredential))
+            {
+                Assert.Fail(string.Format("Expected credential of type ServiceAccountCredential. Obtained {0} instead.", credential.GetType()));
+            }
+        }
+
+        #endregion
+
+        #region Invalid Cases
+
+        // No credential files or environment variable specified - like a fresh developer's machine. 
+        [Test]
+        public void TestGetApplicationDefaultCredentials_NoCredentialFiles()
+        {
+            try
+            {
+                TestDefaultCredentialProvider credentialProvider = new TestDefaultCredentialProvider();
+                ICredential credential = credentialProvider.GetApplicationDefaultCredential();
```

Comment:
In general I'll try to include as much as less in the try block, so you can do the following:
new TestDefaultCredentialProvider().GetApplicationDefaultCredential();

Since you are not using the variable anywhere else.